### DISCUSSION TITLE
Best Practices for Optimizing ORDER BY Performance with Large Datasets

### DISCUSSION THREAD

**Discussion by Ajith-mialo** [getredash/redash#6906] [Q&A]
Hi everyone,

I'm currently working with Redash to query and visualize large datasets, and I'm encountering performance issues when using ORDER BY on tables with millions of rows. The sorting operation seems to be taking a significant amount of time, impacting the overall query performance.

I've already explored some general strategies like indexing and limiting result sets, but I'd like to hear from the community about any additional best practices or specific Redash settings that can help optimize ORDER BY performance for large datasets.

**Comment by justinclift** [comment]
What type of data source are you running the query on? :smile:
